### PR TITLE
`isLt`, `isLte`, `isGt`, `isGte` implemented

### PR DESCRIPTION
**Pull request by niloy**
Implemented `isLt`, `isLte`, `isGt`, `isGte`.

``` js
R.map(R.isLte(3), [1, 2, 3, 4]); // [true, true, true, false]
R.map(R.isLt(3), [1, 2, 3, 4]); // [true, true, false, false]
R.map(R.isGt(3), [1, 2, 3, 4]); // [false, false, false, true]
R.map(R.isGte(3), [1, 2, 3, 4]); // [false, false, true, true]
```

### CODE REVIEW

**Comment by buzzdecafe** [comment]
lgtm. :cow2:

Side note: to my mind, this makes the placeholder redundant.

File: src/isGt.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+var flip = require('./flip');
+var gt = require('./gt');
+
+/**
+ * Returns `true` if the second argument is greater than the first; `false`
+ * otherwise. This function is meant to be used in a curried fashion.
+ *
+ * @func
+ * @memberOf R
+ * @since v0.1.8
```
Comment (svozza) [inline src/isGt.js]: Is that version number right?

Comment (davidchambers) [inline src/isGt.js]: Please don't include `@since` when defining new functions.

Comment (asaf-romano) [inline src/isGt.js]: R.prop(__, OBJ) is quite useful (same goes for contains).

**Comment by asaf-romano** [comment]
@buzzdecafe  R.prop(__, OBJ) is quite useful (same goes for contains).

**Comment by buzzdecafe** [comment]
>  R.prop(__, OBJ) is quite useful (same goes for contains).

maybe so. object mutation can also be useful.

**Comment by CrossEye** [comment]
> Side note: to my mind, this makes the placeholder redundant. 

It doesn't to me -- not quite yet -- as there are still a few non-commutative binary operators that probably also need similar glosses.  `subtract`, `divide`, and `modulo` come immediately to mind.  I might also agree with @asaf-romano about wanting something for `prop` and `contains`, although I'm less certain of those.

I'm not bothered by having to apply `flip` to functions when needed.  It's the binary operators that really _look_ to be lying to you such as `lt(5)` and `subtract(42)` which concern me.

**Comment by niloy** [comment]
@davidchambers Removed `@since`

**Comment by svozza** [comment]
I can count on one hand the amount of times I've used the placeholder for the `gt` or `lt` functions. The main times I use it are for `assoc*`, `difference` and`prop` but mostly `contains`. That's not to mention various custom functions that have been defined with `curry`.

**Comment by davidchambers** [comment]
I'm not sure about this change. Currently we can write `R.gte(_, 18)`, which is fairly clear. With this change we'd be able to write `R.isGte(18)`, which is certainly slightly clearer.

But `R.gte(_, 18)` is *not* the concern, as it's not a source of confusion. `R.gte(18)` is the problem. The existence of `R.isGte` would not prevent `R.gte(18)` from being written, nor from being misread. Instead of a situation in which one might write `R.gte(18)` when one meant to write `R.gte(_, 18)`, we'd be in a situation in which one might write `R.gte(18)` when one meant to write `R.isGte(18)`.

**Comment by niloy** [comment]
@davidchambers Should `R.gte` throw when only 1 argument is passed and suggest to use `R.isGte` instead?

**Comment by buzzdecafe** [comment]
> `R.gte(18)` is the problem.

indeed. this all feels very familiar. This is why placeholders for these non-commutative infix ops make some sense. But since it is difficult to distinguish/communicate which functions were non-commutative the placeholder was generalized. IMO that should be reverted, but i understand i am in the minority in that view. 

> Should `R.gte` throw when only 1 argument is passed and suggest to use `R.isGte` instead?

pardon my butting in, but my $0.02: no, it absolutely should not throw. there is nothing wrong with `R.gte(x)`, and we should give the user the benefit of the doubt.

**Comment by buzzdecafe** [comment]
... and i don't have a better solution for converting these infixes to prefix. I suspect there is not a good, comprehensive solution.

**Comment by davidchambers** [comment]
> Should `R.gte` display console warning when only 1 argument is passed and suggest to use `R.isGte` instead?

That would require currying `R.gte` manually, which would be no fun at all (assuming we'd still support the placeholder).

I wouldn't want the function to throw, in any case. As @paldepind pointed out, one can already write `const gte = R.lt;`, and then use `gte(18)`.

**Comment by CrossEye** [comment]
I'm generally in favor of this, with the caveat that the documentation for `lt`, `gte`, etc. will need some significant warnings about the potential issues and links to `isLt`, `isGte`, etc.

The current situation has the same issue that @davidchambers describes, but I believe that `isGte(18)` is substantially clearer than `gte(_, 18)`, not just slightly clearer.

I'm not even close to ready to give up our "curry everything" ideal, but this is tempting me to push for allowing _only_ these signatures:

```js
    isGte(18)(someVal); // only unary, returns unary
    gte(someVal, 18);  // only binary
```

**Comment by niloy** [comment]
I think unary only `isGte` is a good idea and would like to implement it if others agree. In the meantime, I will update the documentation about the hazards of using single argument to `lt`, `gt` etc.

**Comment by paldepind** [comment]
>  I'm not even close to ready to give up our "curry everything" ideal, but this is tempting me to push for allowing _only_ these signatures:

> ```js
    isGte(18)(someVal); // only unary, returns unary
    gte(someVal, 18);  // only binary
```

I can see the temptation. It is pretty much the same as making `R.gte` throw when given only a single argument and making `R.isGte` throw when given two.

Solving it through documentation is probably a better approach.

**Comment by niloy** [comment]
Added documentation about possible incorrect usage of `R.lt(18)` etc.

**Comment by CrossEye** [comment]
> Solving it through documentation is probably a better approach.

It definitely is.  But the temptation is real, as it would in some way rid us of this problem.

**Comment by CrossEye** [comment]
I believe this didn't get resolved.  Reopening to complete the discussion.

**Comment by niloy** [comment]
Kewl :+1: :smiley:

**Comment by kedashoe** [comment]
I have had `isGt` and `isLt` in my ramda wrapper for forever. I don't mind the placeholder, so it doesn't really bother me to write `R.gt(R.__, x)`. The thing is, I literally never use the comparison functions without the placeholder as the first argument. If I already know all the arguments for my comparison, I just write `x > 10` rather than `R.gt(x, 10)`. Otherwise, I need them as part of some composition pipeline.. which means I never actually use `gt` and `lt` at all, only `isGt` and `isLt`.

As ramda is still pre 1.0, I would rather we just switch the functions we already have than include all 8 comparison functions. But I do use these, so if there is not the stomach for that, I would be +1 for this.

**Comment by tycho01** [comment]
On the placeholder, most examples in favor of it here seem binary functions where `flip` might do?
(I don't have a strong stance in that discussion, was just trying to understand as it seems I'm missing something.)

**Comment by CrossEye** [comment]
Yes, `flip` would do for many cases where the placeholder is used.  But it's not applicable to all.

Closing again.  We still need to address this, but obviously I was overeager in reopening it when I did.  :frowning_face:

**Pull request closed without merging**